lib/commands/deploynodeapp.js

```diff
@@ -425,6 +491,137 @@ function proxyCreationDone(err, req, body, opts, done) {
   }
 }
 
+function importDeployToEdgeServices(opts, request, done) {
+  async.waterfall([
+    function(cb) {
+      uploadNodeSourceToEdgeServices(opts, request, cb);
+    },
+    function(imported, cb) {
+      deployEdgeServicesApplication(opts, request, imported, cb)
+    }
+  ], done)
+}
+
+function uploadNodeSourceToEdgeServices(opts, request, done) {
+  // get a temp file to archive the source code to
+  tmp.file({postfix: '.tgz'}, function(err, target) {
+    if (err) { return done(err) }
+
+    if (opts.debug) { console.log('compressing the source code'); }
+    // archive & compress the source code directory
+    targz.compress(opts.directory, target, function(err) {
+      if (err) { return done(err) }
+
+      var uri = util.format('http://%s/organizations/%s/apps?stream=%s', opts.edgeserviceshost, opts.organization, opts.verbose);
+      
+      if (opts.debug) { console.log('importing the archived source code into EdgeServices'); }
+      // import source code archive into EdgeServices, stream messages
+      request({
+        uri: uri,
+        method: 'POST',
+        json: false,
+        formData: {
+          runtime: 'node',
+          runtimeVersion: opts['node-version'],
+          application: opts.api,
+          file: fs.createReadStream(target)
+        }
+      }).on('error', done)
+      .on('response', function(response) {
+        if (response.statusCode === 400) { // something was wrong in the request
+          return done(new Error('bad request, there was an issue with the request payload'))
+        } else if (response.statusCode !== 202) {
+          return done(new Error(util.format('Received non-OK status code: %d', response.statusCode)))
+        }
+
+        var lastMessage = {};
+        response.on('data', function(data) { // parse chunked transfer of response
+          var msg = JSON.parse(data)
+          if (msg.error !== undefined) {
+            return done(new Error(util.format('Error during application import: %s', msg.error)), msg)
+          }
+
+          if (msg.done) {
+            if (opts.debug || opts.verbose) { console.log(util.format('Application "%s" revision "%s" built for "%s"', opts.api, msg.revision.revision, opts.organization)) }
+            return done(null, msg) // end with the last message passed to the results
+          } else if (opts.verbose || opts.debug) {
+            console.log(msg.message)
+          }
+        })
+      })
+    })
+  })
+}
+
+function deployEdgeServicesApplication(opts, request, imported, cb) {
+  var uri = util.format('http://%s/environments/%s:%s/deployments', opts.edgeserviceshost, opts.organization, opts.environments);
+  var payload = {
+    name: opts.api,
```

Comment:
yeah good point...I don't think we need to expose the configurable deployment name in this CLI.

So I can leave the `name` out of the deployment payload and it'll be set to the application?